src/main/js/controls/Bubble/helpers/helpers.js

```diff
@@ -0,0 +1,544 @@
+ * Processes the input JSON and adds the shapes, colors, labels etc. to each data points so that we
+ * can use them when rendering the data point.
+ *
+ * @private
+ * @param {object} graphConfig - config object of Graph API
+ * @param {object} dataTarget - Data points object
+ * @returns {object} dataTarget - Updated data target object
+ */
+const processDataPoints = (graphConfig, dataTarget) => {
+    const type = graphConfig.axis.x.type;
+    const getXDataValues = (x) => {
+        if (!isValidAxisType(x, type)) {
+            throw new Error(errors.THROW_MSG_INVALID_FORMAT_TYPE);
+        }
+        return parseTypedValue(x, type);
+    };
+
+    graphConfig.shownTargets.push(dataTarget.key);
+    dataTarget.internalValuesSubset = dataTarget.values.map((value) => ({
+        onClick: dataTarget.onClick,
+        isCritical: value.isCritical || false,
+        x: getXDataValues(value.x),
+        y: value.y,
+        weight: value.weight,
+        color: dataTarget.color || constants.DEFAULT_COLOR,
+        label: dataTarget.label || {},
+        yAxis: dataTarget.yAxis || constants.Y_AXIS,
+        key: dataTarget.key
+    }));
+    return dataTarget;
+};
+/**
+ * Returns the internal values subset which is the array that was created from the input JSON.
+ * This array has information for each data point w.r.t shape, colors and on click callback along with
+ * x and y co-ordinates.
+ *
+ * @private
+ * @param {object} target - Object containing the subsets
+ * @returns {Array} List of data point subsets
+ */
+const getDataPointValues = (target) => target.internalValuesSubset;
+/**
+ * Checks the data-set is currently shown in the graph and if the y data-point value is null
+ * If they are then true, false otherwise
+ *
+ * @private
+ * @param {object} shownTargets - graph targets config object
+ * @param {object} value - data point value object
+ * @returns {boolean} true if data point needs to be hidden, false otherwise
+ */
+const shouldHideDataPoints = (shownTargets, value) =>
+    shownTargets.indexOf(value.key) < 0 || value.y === null;
+
+/**
+ * Enforces blur state for all the bubbles that is not the one hovered on.
+ * This is provided regardless of whether onClick is present or not.
+ *
+ * @private
+ * @param {object} target - target node of bubble which is clicked
+ * @returns {undefined} - returns nothing
+ */
+const enforceBubbleBlur = (target) => {
+    d3.selectAll(`.${styles.point}`)
+        .select("circle")
+        .attr("fill-opacity", constants.DEFAULT_BUBBLE_BLUR_OPACITY)
+        .attr("stroke-opacity", constants.DEFAULT_BUBBLE_BLUR_STROKE_OPACITY);
+
+    d3.select(target)
+        .select("circle")
+        .attr("fill-opacity", constants.DEFAULT_BUBBLE_OPACITY)
+        .attr("stroke-opacity", constants.DEFAULT_BUBBLE_STROKE_OPACITY);
+};
+
+/**
+ * Removes the carbon-bubbleBlur style from all the bubbles to unblur all the bubbles in the bubble graph.
+ *
+ * @private
+ * @returns {undefined} - returns nothing
```

Comment:
`@returns {object} - d3 Selection object`

Reply:
done in the latest commit